### ISSUE TITLE
Failed to execute 'postMessage' on 'DedicatedWorkerGlobalScope'

### ISSUE THREAD

**Issue by achmadk** [andywer/threads.js#192]
Hello, @andywer , thanks to develop `threads.js`. I want to use it for implementing web worker into my web app.

I want to move redux offline store initialization into web worker using `threads.js`. Although it can be achieved in [this tutorial](https://dassur.ma/things/react-redux-comlink/) with `comlink` and `rollup`, I want to implement it with `webpack` and `threads.js`.

Unfortunately, I got error like this
```sh
Error: Failed to execute 'postMessage' on 'DedicatedWorkerGlobalScope': function (action) {
        if (typeof action === 'function') {
          return action(dispatch, getState, ext...<omitted>... } could not be cloned.
    at Object.postMessageToMaster (implementation.browser.js:9)
    at postJobResultMessage (index.js:73)
    at eval (index.js:107)
    at Generator.next (<anonymous>)
    at fulfilled (index.js:5)
``` 

If I dont use `redux-thunk` as one of redux middleware, I still get same error with different message
```sh
Error: Failed to execute 'postMessage' on 'DedicatedWorkerGlobalScope': function dispatch(action) {
    if (!isPlainObject(action)) {
      throw new Error('Actions must be plain obje...<omitted>... } could not be cloned.
    at Object.postMessageToMaster (implementation.browser.js:9)
    at postJobResultMessage (index.js:73)
    at eval (index.js:107)
    at Generator.next (<anonymous>)
    at fulfilled (index.js:5)
```

Here is source codes of worker
```js
import { createStore, applyMiddleware } from 'redux'
import { persistStore } from 'redux-persist'
import { composeWithDevTools } from 'redux-devtools-extension'
import thunk from 'redux-thunk'
import { expose } from 'threads/worker'

import { appReducer } from './reducers'

const composeEnhancer = composeWithDevTools({})

const appMiddleware = applyMiddleware(thunk);

expose(function initStore () {
  const store = createStore(
    appReducer,
    process.env.NODE_ENV === 'production'
      ? compose(appMiddleware)
      : composeEnhancer(appMiddleware)
    )
  const persistor = persistStore(store)
  return {
    store,
    persistor
  }
})

```

And here is source codes for main thread (`spawn` and `Worker` has been imported)
```js
async componentDidMount () {
    try {
      const initStore = await spawn(new Worker("../path/to/worker"));
      const { store, persistor } = await initStore()
      this.setState({ store, persistor })
      await initLiff();
    } catch (error) {
      console.log(error)
    }
  }
```

**Comment by andywer** [comment]
Thanks, @achmadk!

So the issue here is that you try to return an object containing functions from the worker which doesn't work, since functions are not serializable by the browser, at least not like that. This is also a limitation that you would need to face with Comlink, for instance.

I think the best solution here is to eagerly initialize the store and `expose({ dispatch: store.dispatch })`.

PS: Using plain standard actions should be perfectly fine, but you won't be able to pass thunks (as in `redux-thunk`), since you will run into the same `postMessage()` problem. There is a solution to that coming up, though: #145

**Comment by andywer** [comment]
Let's discuss in #145 if it's about passing functions. There was also some sub-optimal error handling involved that should have been fixed by #202.

Sorry for the delay, btw. I was sick recently.
